docs/zkapps/o1js/common-types-and-functions.md

```diff
@@ -0,0 +1,354 @@
+---
+title: o1js Common Types and Functions
+hide_title: true
+sidebar_label: Common Types and Functions
+description: o1js includes a lot of built in types and functions for you to start creating your zkApp right away.
+keywords:
+  - zero knowledge proof programming
+  - zk proof
+  - zk
+  - data types
+  - o1js
+  - blockchain
+  - mina
+  - typescript
+  - methods
+  - common types
+  - built in types
+  - bool
+  - uint32
+  - uint64
+  - provable if
+  - loops
+  - functions
+---
+
+:::info
+
+zkApp programmability is not yet available on the Mina Mainnet. You can get started now by deploying zkApps to the Berkeley Testnet.
+
+:::
+
+# o1js Built-in Data Types
+
+o1js includes built in types and methods that you can use inside your zkApp and ZKPs (zero knowledge proofs). All o1js provable types are built using the [Field](./basic-concepts.md#field) type, each with unique properties and advantages. You can find detailed class description in [o1js reference](../o1js-reference).
+
+## Bool
+
+`Bool` is the type used for `boolean` values. It is equivalent to the `boolean` type in order languages and frameworks.
+
+<!-- prettier-ignore -->
+```ts
+const x = new Bool(true);
+const y = new Bool(false);
+```
+
+With the `Bool` type, you can call the contructor without `new`, just like the `Field` type. The best practice is to use the constructor function without `new`:
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(false);
+```
+
+You can also create a `Bool` from another `Bool`.
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(x);
+```
+
+It is also possible to convert a `Field` into a `Bool`. However, you should make sure that the `Field` is "bool-like" (with a value either 1 or 0) before converting it.
+
+```ts
+let x = Field(1);
+x.assertBool(); // Asserts that x has a value either 1 or 0
+
+let x_bool = Bool(x.value);
```

Comment:
This needs to be replaced with `Bool.Unsafe.ofField`. it's important that this API is _explicit_ about its unsafety. The API you show is internal and it's even problematic that it type checks

Reply:
Can I directly use `x.equals(Field(1)).or(x.equals(Field(0))).assertTrue()`, maybe that would be better than giving the internal API as an example? Or else I can directly delete this section, but I think it is important to show how conversions should work in o1js to new comers. Thank you @mitschabaude

Reply:
I think it's useful to show the conversion. Just use
```ts
let xBool = Bool.Unsafe.ofField(x);
```